### DISCUSSION TITLE
Have an API with Authentication

### DISCUSSION THREAD

**Discussion by sugizo** [directus/directus#5342] [Archive]
**Objective**
Have an API with Authentication

**Example in terminal**
**HTTP Header Jwt**
`curl -X GET -i http://localhost:8055/items/instrument/ -H "Authorization: paste_jwt_token_here"`

**HTTP Header Key**
`curl -X GET -i http://localhost:8055/items/instrument/ -H 'custom_api_key: whatever'`

**HTTP Basic Auth**
`curl -X GET -d username=admin -d password=password -i http://localhost:8055/items/instrument/`

**Result**
tried from scratch tested everything above command is work (can return a value), but it's seems not meet with expectation

**Expectation**
Before the action CRUD the API must authenticated and authorized the user either with JWT, custom API Key in header or basic Authentication

**Question**
how to achieve it using directus way?

thanks and best regards

**Comment by rijkvanzanten** [comment]
See https://docs.directus.io/reference/api/authentication/

**Reply by sugizo** [reply]
@rijkvanzanten thanks for the link, from that link learn how to achieve static token (after define token on directus users) with : 
**e.g.**
```
curl -X GET -i http://localhost:8055/items/instrument/1?access_token=access_token
curl -X GET -i http://localhost:8055/items/instrument/1 -H "Authorization: Bearer access_token"
```
but
my inital question still not answered how to authenticate directus with : 
**custom header**
and
**http basic login**

for http basic login already tried
```
curl -X POST -d email=admin@example.com -d password=password -i http://localhost:8055/auth/login/
curl -X POST -d '{"email": "admin@example.com", "password": "password"}' -i http://localhost:8055/auth/login
curl -X POST -d "email":"admin@example.com" -d "password":"password" -i http://localhost:8055/auth/login
curl -X POST -d "email"="admin@example.com" -d "password"="password" -i http://localhost:8055/auth/login
curl -X POST -d "email=admin@example.com&password=password" -i http://localhost:8055/auth/login
```
**result**
`{"errors":[{"message":"\"email\" is required","extensions":{"code":"INVALID_PAYLOAD"}}]}`

**2nd trial**
`curl -X GET -d email=admin@example.com -d password=password -i http://localhost:8055/items/instrument/1`
**result**
`{"errors":[{"message":"You don't have permission to access this.","extensions":{"code":"FORBIDDEN"}}]}`

**3rd trial**
`curl -X POST -d email=admin@example.com -d password=password -i http://localhost:8055/items/instrument/1`
**result**
`{"errors":[{"message":"Route /items/instrument/1 doesn't exist.","extensions":{"code":"ROUTE_NOT_FOUND"}}]}`

# Question
how to authenticate directus with : 
**custom header**
and
**http basic login**
?

thanks and best regards

**Reply by rijkvanzanten** [reply]
> how to authenticate directus with : custom header and http basic login?

That's not what you asked in your OP though 😁 

We don't support custom headers for auth, nor do we support basic auth (for security reasons).
